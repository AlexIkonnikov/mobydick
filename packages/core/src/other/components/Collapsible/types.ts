import {
  LayoutAnimationProperty,
  LayoutAnimationType,
  TextStyle,
  TouchableOpacityProps,
  ViewStyle,
} from 'react-native';
import {ReactChild} from 'react';

import {TypographyProp} from '../../../typography';

export interface ICollapsibleProps extends TouchableOpacityProps {
  title?: string;
  children?: ReactChild;
  duration?: number;
  containerStyle?: ViewStyle | ViewStyle[];
  headerStyle?: ViewStyle | ViewStyle[];
  titleStyle?: TextStyle | TextStyle[];
  fontTitle?: TypographyProp;
  typeAnimation?: LayoutAnimationType;
  creationPropAnimation?: LayoutAnimationProperty;
  numberOfLines?: number;
}
