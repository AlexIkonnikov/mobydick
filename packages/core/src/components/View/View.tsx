import React, {FC} from 'react';
import {View as DefaultView} from 'react-native';

import {ViewProps} from './types';

const TextInput: FC<ViewProps> = props => {
  return <DefaultView {...props} />;
};

export default TextInput;
