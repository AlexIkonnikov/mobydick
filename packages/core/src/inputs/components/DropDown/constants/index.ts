import rem from '../../../../styles/utils/rem';

export const BORDER_BUTTON_WIDTH = rem(1);

export const DROP_DOWN_POPUP_ID = 'DropDownPopup';

export const DEFAULT_DROP_DOWN_HEIGHT = rem(48);
export const DEFAULT_DROP_DOWN_WIDTH = rem(200);
export const DEFAULT_DROPDOWN_MARGIN_FROM_BUTTON = rem(4);

export const LIST_MAX_HEIGHT = rem(244);
export const EMPTY_LIST_HEIGHT = rem(74);
