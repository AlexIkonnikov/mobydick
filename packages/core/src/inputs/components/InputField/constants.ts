enum Constants {
  testID = 'TextInput',
}

export default Constants;
