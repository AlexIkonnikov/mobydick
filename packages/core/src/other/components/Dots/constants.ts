import rem from '../../../styles/utils/rem';

export const SPAN_SIZE = 3;

export const SIZE_LARGE = rem(8);
export const SIZE_MEDIUM = rem(6);
export const SIZE_SMALL = rem(4);

export const MARGIN_DOT = rem(5) * 2;

export const WIDTH_SMALL =
  (SIZE_LARGE + MARGIN_DOT) * 3 +
  (SIZE_MEDIUM + MARGIN_DOT) +
  (SIZE_SMALL + MARGIN_DOT);

export const WIDTH_MEDIUM = WIDTH_SMALL + (SIZE_MEDIUM + MARGIN_DOT);

export const WIDTH_LARGE = WIDTH_MEDIUM + (SIZE_SMALL + MARGIN_DOT);
