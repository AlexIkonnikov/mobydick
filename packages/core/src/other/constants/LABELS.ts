export const LABELS = {
  togglePress: 'togglePress',
  slider: 'slider',
  sliderLayoutLowThumb: 'sliderLayoutLowThumb',
};
