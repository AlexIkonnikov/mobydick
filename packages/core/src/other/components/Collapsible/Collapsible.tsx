import React, {useCallback, useMemo, useState} from 'react';
import {LayoutAnimation, Platform, StyleSheet, UIManager} from 'react-native';

import {Typography} from '../../../typography';
import TouchableOpacity from '../../../basic/components/TouchableOpacity/TouchableOpacity';
import SimpleIcon from '../../../styles/icons/font/SimpleIcon';
import View from '../../../basic/components/View/View';
import {IThemeContext} from '../../../styles';
import useTheme from '../../../styles/theme/hooks/useTheme';
import useStyles from '../../../styles/theme/hooks/useStyles';
import {LABELS} from '../../constants';

import {ICollapsibleProps} from './types';

if (
  Platform.OS === 'android' &&
  UIManager.setLayoutAnimationEnabledExperimental
) {
  UIManager.setLayoutAnimationEnabledExperimental(true);
}

const Collapsible = (props: ICollapsibleProps) => {
  const {
    title,
    children,
    duration = 250,
    containerStyle,
    fontTitle = 'SemiBold-Secondary-M',
    headerStyle,
    titleStyle,
    typeAnimation = 'easeInEaseOut',
    creationPropAnimation = 'scaleY',
    numberOfLines = 2,
    initialState = false,
    isAnimated = true,
    titleBottomView,
  } = props;
  const [styles] = useStyles(createStyles);
  const {colors} = useTheme();
  const [collapsed, setCollapsed] = useState(initialState);

  const onPress = useCallback(() => {
    setCollapsed(!collapsed);
    isAnimated &&
      LayoutAnimation.configureNext(
        LayoutAnimation.create(duration, typeAnimation, creationPropAnimation),
      );
  }, [collapsed, duration, isAnimated]);

  const name = useMemo(() => {
    return collapsed ? 'icon-arrow-down' : 'icon-arrow-right';
  }, [collapsed]);

  return (
    <View style={[styles.container, containerStyle]}>
      <TouchableOpacity
        onPress={onPress}
        style={[styles.header, headerStyle]}
        accessibilityLabel={LABELS.collapsed}>
        <Typography
          font={fontTitle}
          numberOfLines={numberOfLines}
          style={[styles.title, titleStyle]}>
          {title}
        </Typography>
        <SimpleIcon name={name} color={colors.IconNeutral} />
      </TouchableOpacity>
      {titleBottomView}
      {collapsed && children}
    </View>
  );
};

const createStyles = ({spaces}: IThemeContext) =>
  StyleSheet.create({
    container: {
      width: '100%',
      padding: spaces.Space20,
    },
    header: {
      flexDirection: 'row',
      alignItems: 'center',
      justifyContent: 'space-between',
    },
    title: {
      flex: 1,
    },
  });

export default Collapsible;
