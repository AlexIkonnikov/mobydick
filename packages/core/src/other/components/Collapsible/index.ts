import Collapsible from './Collapsible';

export {Collapsible};

export * from './types';
