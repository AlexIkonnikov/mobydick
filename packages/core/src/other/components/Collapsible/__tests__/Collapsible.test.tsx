import {fireEvent, render} from '@testing-library/react-native';
import React from 'react';
import {Platform} from 'react-native';

import Collapsible from '../Collapsible';
import View from '../../../../basic/components/View/View';
import {LABELS} from '../../../constants';

describe('Collapsible', () => {
  test('render crossed onPress - 1', () => {
    const {toJSON, getByLabelText} = render(
      <Collapsible
        title={'title'}
        duration={1000}
        containerStyle={{backgroundColor: 'red'}}
        headerStyle={{backgroundColor: 'red'}}
        typeAnimation={'linear'}
        creationPropAnimation={'opacity'}
        fontTitle={'Regular-Black-S'}
        titleStyle={{flex: 1}}
        numberOfLines={1}>
        <>
          <View />
        </>
      </Collapsible>,
    );

    const onPress = getByLabelText(LABELS.collapsed);

    fireEvent.press(onPress);

    expect(toJSON()).toMatchSnapshot();
  });
  test('render crossed onPress - 1 android', () => {
    Platform.OS = 'android';

    const {toJSON, getByLabelText} = render(
      <Collapsible title={'title'} duration={1000}>
        <>
          <View />
        </>
      </Collapsible>,
    );

    const onPress = getByLabelText(LABELS.collapsed);

    fireEvent.press(onPress);

    expect(toJSON()).toMatchSnapshot();
  });
  test('render crossed onPress - 2', () => {
    const {toJSON, getByLabelText} = render(
      <Collapsible title={'title'} initialState={true} isAnimated={false}>
        <>
          <View />
        </>
      </Collapsible>,
    );

    const onPress = getByLabelText(LABELS.collapsed);

    fireEvent.press(onPress);
    fireEvent.press(onPress);

    expect(toJSON()).toMatchSnapshot();
  });
});
