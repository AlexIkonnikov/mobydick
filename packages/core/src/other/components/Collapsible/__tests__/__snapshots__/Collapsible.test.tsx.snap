// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapsible render crossed onPress - 1 1`] = `
<View
  style={
    [
      {
        "padding": 30,
        "width": "100%",
      },
      {
        "backgroundColor": "red",
      },
    ]
  }
>
  <View
    accessibilityLabel="collapsed"
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "red",
        "flexDirection": "row",
        "height": 66,
        "justifyContent": "space-between",
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={true}
      numberOfLines={1}
      style={
        [
          {
            "color": "#20242D",
            "fontFamily": "Inter-Regular",
            "fontSize": 22.5,
            "lineHeight": 28.5,
          },
          [
            {
              "flex": 1,
            },
            {
              "flex": 1,
            },
          ],
        ]
      }
    >
      title
    </Text>
    <Text
      allowFontScaling={true}
      style={
        {
          "color": "#5E6678",
          "fontFamily": "Neotis",
          "fontSize": 36,
        }
      }
    >
      
    </Text>
  </View>
  <View />
</View>
`;

exports[`Collapsible render crossed onPress - 1 android 1`] = `
<View
  style={
    [
      {
        "padding": 30,
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <View
    accessibilityLabel="collapsed"
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 66,
        "justifyContent": "space-between",
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={true}
      numberOfLines={2}
      style={
        [
          {
            "color": "#444B5A",
            "fontFamily": "Inter-SemiBold",
            "fontSize": 24,
            "lineHeight": 30,
          },
          [
            {
              "flex": 1,
            },
            undefined,
          ],
        ]
      }
    >
      title
    </Text>
    <Text
      allowFontScaling={true}
      style={
        {
          "color": "#5E6678",
          "fontFamily": "Neotis",
          "fontSize": 36,
        }
      }
    >
      
    </Text>
  </View>
  <View />
</View>
`;

exports[`Collapsible render crossed onPress - 2 1`] = `
<View
  style={
    [
      {
        "padding": 30,
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <View
    accessibilityLabel="collapsed"
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 66,
        "justifyContent": "space-between",
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={true}
      numberOfLines={2}
      style={
        [
          {
            "color": "#444B5A",
            "fontFamily": "Inter-SemiBold",
            "fontSize": 24,
            "lineHeight": 30,
          },
          [
            {
              "flex": 1,
            },
            undefined,
          ],
        ]
      }
    >
      title
    </Text>
    <Text
      allowFontScaling={true}
      style={
        {
          "color": "#5E6678",
          "fontFamily": "Neotis",
          "fontSize": 36,
        }
      }
    >
      
    </Text>
  </View>
</View>
`;
